src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1065,11 +1065,19 @@ def config_diagnostics(resource_group_name, name, level=None,
         application_logs = ApplicationLogsConfig(fs_log)
 
     http_logs = None
-    if web_server_logging is not None:
-        enabled = web_server_logging
-        # 100 mb max log size, retenting last 3 days. Yes we hard code it, portal does too
-        fs_server_log = FileSystemHttpLogsConfig(100, 3, enabled)
-        http_logs = HttpLogsConfig(fs_server_log)
+    logs_enabled = (web_server_logging if web_server_logging is not None
+                    else docker_container_logging)
+
+    # 100 mb max log size, retenting last 3 days. Yes we hard code it, portal does too
+    filesystem_log_config = FileSystemHttpLogsConfig(100, 3, logs_enabled == 'filesystem')
+    az_blob_storage_log_config = AzureBlobStorageHttpLogsConfig(None, 3, False)
+
+    if logs_enabled == 'storage':
+        if sas_url is None:
+            raise CLIError("sas url cannot be null if storage is selected!")
+        az_blob_storage_log_config = AzureBlobStorageHttpLogsConfig(sas_url, 3, True)
```

Comment:
nit: this logic can be put in a validator.

Reply:
Not sure what you mean. Is there an example of this?

Reply:
Oh. Sorry for not providing the background. "Validators" are the function attached to parameter where the parameter as well as the entire namespace, which is the argparse term for the collection of all user input, can be validated or modified.

Please check the `_validators.py` or `validators.py` file in various command modules as examples.

This is not a requirement.